hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/AbstractCSQueue.java

```diff
@@ -1091,7 +1092,8 @@ Resource getQueueMaxResource(String nodePartition) {
     return getEffectiveMaxCapacity(nodePartition);
   }
 
-  public boolean hasChildQueues() {
```

Comment:
As @TaoYang526 also pointed out in the jira, this is also being used by production code so the VisibleForTesting annotation is not applicable for this one.

Reply:
As a second thought: hasChildQueues is called from org.apache.hadoop.yarn.server.resourcemanager.scheduler.capacity.AbstractCSQueue#canAssignToThisQueue so technically it could have been private if it wasn't accessed from test code so I think VisibleForTesting is correct here.